actionpack/lib/action_controller/metal/content_security_policy.rb

```diff
@@ -28,9 +28,19 @@ def content_security_policy(enabled = true, **options, &block)
         end
       end
 
-      def content_security_policy_report_only(report_only = true, **options)
+      def content_security_policy_report_only(enabled = true, **options, &block)
         before_action(options) do
-          request.content_security_policy_report_only = report_only
+          if block_given?
+            policy = current_content_security_policy
+            yield policy
+            request.content_security_policy_report_only = policy
+          else
+            request.content_security_policy_report_only = true
```

Comment:
this is to support usage in a controller without a block or assignment

```
def action
   content_security_policy_report_only
end
```

Reply:
This support is useful! Is there a test that demonstrates it exists?

Reply:
Yep, it is existing functionality covered here: https://github.com/rails/rails/blob/9292db856969283eff82509f0ea0537cc039c382/actionpack/test/dispatch/content_security_policy_test.rb#L362